lib/src/components/NxFileUpload/__tests__/NxFileUpload.test.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { pipe } from 'ramda';
+
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+import NxFileUpload from '../NxFileUpload';
+
+describe('NxFileUpload', function() {
+  const minimalProps = {
+        files: null,
+        onChange: () => {}
+      },
+      render = rtlRender(NxFileUpload, minimalProps),
+      renderEl = rtlRenderElement(NxFileUpload, minimalProps),
+      renderInput = pipe(renderEl, el => el?.querySelector('input[type=file]'));
```

Comment:
For clarity, renderInput is a function that takes the returned value of renderEl (which is the render of NxFileUpload, inside of the render() method's container), and then just querySelecting within NxFileUpload to find the input?

Reply:
> For clarity, renderInput is a function that takes the returned value of renderEl (which is the render of NxFileUpload, inside of the render() method's container), and then just querySelecting within NxFileUpload to find the input?

Yes. So overall, you pass it any extra props you want, and it returns the `<input>` of the resulting `NxFileUpload`.

Reply:
Great, thanks for confirming that.